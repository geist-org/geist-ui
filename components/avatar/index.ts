import Avatar from './avatar'
import AvatarGroup from './avatar-group'

export type AvatarComponentType = typeof Avatar & {
  Group: typeof AvatarGroup
}
;(Avatar as AvatarComponentType).Group = AvatarGroup

export type { AvatarProps } from './avatar'
export type { AvatarGroupProps } from './avatar-group'

export default Avatar as AvatarComponentType
