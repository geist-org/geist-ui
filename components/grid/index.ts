import Grid from './grid'
import GridContainer from './grid-container'

export type GridComponentType = typeof Grid & {
  Container: typeof GridContainer
}
;(Grid as GridComponentType).Container = GridContainer

export type { GridContainerProps } from './grid-container'
export type { GridProps } from './grid'
export type { GridBreakpointsValue } from './basic-item'
export type {
  GridAlignContent,
  GridAlignItems,
  GridDirection,
  GridJustify,
  GridWrap,
} from './grid-types'
export default Grid as GridComponentType
