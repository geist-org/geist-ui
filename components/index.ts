// @ts-ignore
/// <reference types="styled-jsx" />

export { default as AutoComplete } from './auto-complete'
export type { AutoCompleteProps } from './auto-complete'

export { default as Avatar } from './avatar'
export type { AvatarProps, AvatarGroupProps } from './avatar'

export { default as Badge } from './badge'
export type { BadgeProps, BadgeAnchorProps } from './badge'

export { default as Breadcrumbs } from './breadcrumbs'
export type {
  BreadcrumbsProps,
  BreadcrumbsItemProps,
  BreadcrumbsSeparatorProps,
} from './breadcrumbs'

export { default as Button } from './button'
export type { ButtonProps } from './button'

export { default as ButtonDropdown } from './button-dropdown'
export type { ButtonDropdownProps, ButtonDropdownItemProps } from './button-dropdown'

export { default as ButtonGroup } from './button-group'
export type { ButtonGroupProps } from './button-group'

export { default as Capacity } from './capacity'
export type { CapacityProps } from './capacity'

export { default as Card } from './card'
export type { CardProps, CardContentProps, CardFooterProps } from './card'

export { default as Checkbox } from './checkbox'
export type { CheckboxProps, CheckboxGroupProps } from './checkbox'

export { default as Code } from './code'
export type { CodeProps } from './code'

export { default as Collapse } from './collapse'
export type { CollapseProps, CollapseGroupProps } from './collapse'

export { default as Description } from './description'
export type { DescriptionProps } from './description'

export { default as Display } from './display'
export type { DisplayProps } from './display'

export { default as Divider } from './divider'
export type { DividerProps } from './divider'

export { default as Dot } from './dot'
export type { DotProps } from './dot'

export { default as Drawer } from './drawer'
export type { DrawerProps } from './drawer'

export { default as Fieldset } from './fieldset'
export type {
  FieldsetProps,
  FieldsetTitleProps,
  FieldsetSubtitleProps,
  FieldsetGroupProps,
  FieldsetFooterProps,
  FieldsetContentProps,
} from './fieldset'

export { default as GeistProvider } from './geist-provider'
export type { GeistProviderProps } from './geist-provider'

export { default as Grid } from './grid'
export type { GridProps, GridContainerProps } from './grid'

export { default as Image } from './image'
export type { ImageProps, ImageBrowserProps } from './image'

export { default as Input } from './input'
export type { InputProps, InputPasswordProps } from './input'

export { default as Keyboard } from './keyboard'
export type { KeyboardProps } from './keyboard'

export { default as Link } from './link'
export type { LinkProps } from './link'

export { default as Loading } from './loading'
export type { LoadingProps } from './loading'

export { default as Modal } from './modal'
export type {
  ModalProps,
  ModalTitleProps,
  ModalSubtitleProps,
  ModalContentProps,
  ModalActionProps,
} from './modal'

export { default as Note } from './note'
export type { NoteProps } from './note'

export { default as Page } from './page'
export type {
  PageProps,
  PageHeaderProps,
  PageContentProps,
  PageFooterProps,
} from './page'

export { default as Pagination } from './pagination'
export type {
  PaginationProps,
  PaginationNextProps,
  PaginationPreviousProps,
} from './pagination'

export { default as Popover } from './popover'
export type { PopoverProps, PopoverItemProps } from './popover'

export { default as Progress } from './progress'
export type { ProgressProps } from './progress'

export { default as Radio } from './radio'
export type { RadioProps, RadioGroupProps, RadioDescriptionProps } from './radio'

export { default as Rating } from './rating'
export type { RatingProps } from './rating'

export { default as Select } from './select'
export type { SelectProps, SelectOptionProps } from './select'

export { default as Slider } from './slider'
export type { SliderProps } from './slider'

export { default as Snippet } from './snippet'
export type { SnippetProps } from './snippet'

export { default as Spacer } from './spacer'
export type { SpacerProps } from './spacer'

export { default as Spinner } from './spinner'
export type { SpinnerProps } from './spinner'

export { default as Table } from './table'
export type { TableProps, TableColumnProps } from './table'

export { default as Tabs } from './tabs'
export type { TabsProps } from './tabs'

export { default as Tag } from './tag'
export type { TagProps } from './tag'

export { default as Text } from './text'
export type { TextProps } from './text'

export { default as Textarea } from './textarea'
export type { TextareaProps } from './textarea'

export { default as Themes } from './themes'
export type { GeistUIThemes, GeistUserTheme } from './themes'

export { default as Toggle } from './toggle'
export type { ToggleProps } from './toggle'

export { default as Tooltip } from './tooltip'
export type { TooltipProps } from './tooltip'

export { default as Tree } from './tree'
export type { TreeProps } from './tree'

export { default as useAllThemes } from './use-all-themes'
export type { AllThemesConfig } from './use-all-themes'

export { default as useToasts } from './use-toasts'
export type { Toast, ToastInput, ToastAction, ToastLayout } from './use-toasts'

export { default as User } from './user'
export type { UserProps } from './user'

export { default as useBodyScroll } from './use-body-scroll'
export type { BodyScrollOptions } from './use-body-scroll'

export { default as useClipboard } from './use-clipboard'
export type { UseClipboardOptions } from './use-clipboard'

export { default as useMediaQuery } from './use-media-query'
export type { ResponsiveOptions, ResponsiveBreakpoint } from './use-media-query'

export { default as useKeyboard, KeyMod, KeyCode } from './use-keyboard'
export type { KeyboardOptions, UseKeyboardHandler } from './use-keyboard'

export { default as useInput } from './use-input'
export { default as useModal } from './use-modal'
export { default as useTabs } from './use-tabs'
export { default as useClickAway } from './use-click-away'
export { default as useCurrentState } from './use-current-state'
export { default as CssBaseline } from './css-baseline'
export { default as useTheme } from './use-theme'
export { default as useClasses } from './use-classes'
export { default as useScale } from './use-scale'
export { withScale, ScalePropKeys, ScaleContext } from './use-scale'
export type {
  ScaleProps,
  ScaleConfig,
  GetScalePropsFunction,
  GetAllScalePropsFunction,
} from './use-scale'

export { useRect } from './utils/layouts'
export type { ReactiveDomReact } from './utils/layouts'
export { default as Highlight } from './shared/highlight'
export type { HighlightProps } from './shared/highlight'
