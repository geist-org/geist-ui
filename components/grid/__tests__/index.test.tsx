import React from 'react'
import { mount } from 'enzyme'
import { Grid } from 'components'

describe('Grid', () => {
  it('should render correctly', () => {
    const wrapper = mount(
      <Grid.Container>
        <Grid>test</Grid>
        <Grid>test</Grid>
      </Grid.Container>,
    )

    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('all breakpoint values should be supported', () => {
    const wrapper = mount(
      <Grid.Container xs={1} sm={2} md={3} lg={4} xl={5}>
        <Grid xs={1} sm={2} md={3} lg={4} xl={5}>
          test
        </Grid>
      </Grid.Container>,
    )

    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('css value should be passed through', () => {
    const wrapper = mount(
      <Grid.Container
        justify="center"
        alignItems="center"
        alignContent="center"
        direction="column"
        wrap="wrap">
        <Grid
          justify="center"
          alignItems="center"
          alignContent="center"
          direction="column">
          test
        </Grid>
      </Grid.Container>,
    )

    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('decimal spacing should be supported', () => {
    const wrapper = mount(
      <Grid.Container gap={0.11123}>
        <Grid>test</Grid>
      </Grid.Container>,
    )

    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('nested components should be supported', () => {
    const wrapper = mount(
      <Grid.Container>
        <Grid>test</Grid>
        <Grid.Container>
          <Grid>test</Grid>
          <Grid.Container>
            <Grid>test</Grid>
            <Grid.Container>
              <Grid>test</Grid>
            </Grid.Container>
            ,
          </Grid.Container>
          ,
        </Grid.Container>
        ,
      </Grid.Container>,
    )

    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('there should be no alignment class when not set', () => {
    let wrapper = mount(<Grid.Container justify="flex-end" />)
    expect(wrapper.find('.item').hasClass('justify')).toBeTruthy()
    wrapper = mount(<Grid.Container />)
    expect(wrapper.find('.item').hasClass('justify')).not.toBeTruthy()
  })

  it('there should be no responsive class when not set', () => {
    let wrapper = mount(<Grid.Container sm={2} />)
    expect(wrapper.find('.item').hasClass('sm')).toBeTruthy()
    expect(wrapper.find('.item').hasClass('xs')).not.toBeTruthy()
    wrapper = mount(<Grid.Container />)
    expect(wrapper.find('.item').hasClass('sm')).not.toBeTruthy()
  })

  it('should work correctly when use alone', () => {
    const wrapper = mount(<Grid />)
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should work correctly when size exceeds', () => {
    const wrapper = mount(
      <Grid.Container>
        <Grid xs={25}>test</Grid>
        <Grid xs={-1}>test</Grid>
      </Grid.Container>,
    )

    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('Grid should be hidden when value is 0', () => {
    let wrapper = mount(<Grid.Container xs={0} />)
    expect(wrapper.find('.item').hasClass('xs')).toBeTruthy()
    expect(wrapper.find('.item').html()).toContain('display: none')
  })
})
