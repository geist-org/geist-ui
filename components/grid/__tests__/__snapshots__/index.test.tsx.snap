// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid all breakpoint values should be supported 1`] = `
"<div class=\\"item xs sm md lg xl mock \\"><div class=\\"item xs sm md lg xl mock \\">test<style>
        .item {
        }

        .justify {
          justify-content: undefined;
        }

        .direction {
          flex-direction: undefined;
        }

        .alignContent {
          align-content: undefined;
        }

        .alignItems {
          align-items: undefined;
        }

        .xs {
          flex-grow: 0;
          max-width: 4.166666666666667%;
          flex-basis: 4.166666666666667%;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 0;
            max-width: 4.166666666666667%;
            flex-basis: 4.166666666666667%;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 0;
            max-width: 8.333333333333334%;
            flex-basis: 8.333333333333334%;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 0;
            max-width: 12.5%;
            flex-basis: 12.5%;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 0;
            max-width: 16.666666666666668%;
            flex-basis: 16.666666666666668%;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 0;
            max-width: 20.833333333333336%;
            flex-basis: 20.833333333333336%;
            
          }
        }
      </style></div><style>
        .item {
        }

        .justify {
          justify-content: undefined;
        }

        .direction {
          flex-direction: undefined;
        }

        .alignContent {
          align-content: undefined;
        }

        .alignItems {
          align-items: undefined;
        }

        .xs {
          flex-grow: 0;
          max-width: 4.166666666666667%;
          flex-basis: 4.166666666666667%;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 0;
            max-width: 4.166666666666667%;
            flex-basis: 4.166666666666667%;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 0;
            max-width: 8.333333333333334%;
            flex-basis: 8.333333333333334%;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 0;
            max-width: 12.5%;
            flex-basis: 12.5%;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 0;
            max-width: 16.666666666666668%;
            flex-basis: 16.666666666666668%;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 0;
            max-width: 20.833333333333336%;
            flex-basis: 20.833333333333336%;
            
          }
        }
      </style></div>"
`;

exports[`Grid css value should be passed through 1`] = `
"<div class=\\"item justify direction alignItems alignContent mock \\"><div class=\\"item justify direction alignItems alignContent mock \\">test<style>
        .item {
        }

        .justify {
          justify-content: center;
        }

        .direction {
          flex-direction: column;
        }

        .alignContent {
          align-content: center;
        }

        .alignItems {
          align-items: center;
        }

        .xs {
          flex-grow: 1;
          max-width: 100%;
          flex-basis: 0;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }
      </style></div><style>
        .item {
        }

        .justify {
          justify-content: center;
        }

        .direction {
          flex-direction: column;
        }

        .alignContent {
          align-content: center;
        }

        .alignItems {
          align-items: center;
        }

        .xs {
          flex-grow: 1;
          max-width: 100%;
          flex-basis: 0;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }
      </style></div>"
`;

exports[`Grid decimal spacing should be supported 1`] = `
"<div class=\\"item  mock \\"><div class=\\"item  mock \\">test<style>
        .item {
        }

        .justify {
          justify-content: undefined;
        }

        .direction {
          flex-direction: undefined;
        }

        .alignContent {
          align-content: undefined;
        }

        .alignItems {
          align-items: undefined;
        }

        .xs {
          flex-grow: 1;
          max-width: 100%;
          flex-basis: 0;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }
      </style></div><style>
        .item {
        }

        .justify {
          justify-content: undefined;
        }

        .direction {
          flex-direction: undefined;
        }

        .alignContent {
          align-content: undefined;
        }

        .alignItems {
          align-items: undefined;
        }

        .xs {
          flex-grow: 1;
          max-width: 100%;
          flex-basis: 0;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }
      </style></div>"
`;

exports[`Grid nested components should be supported 1`] = `
"<div class=\\"item  mock \\"><div class=\\"item  mock \\">test<style>
        .item {
        }

        .justify {
          justify-content: undefined;
        }

        .direction {
          flex-direction: undefined;
        }

        .alignContent {
          align-content: undefined;
        }

        .alignItems {
          align-items: undefined;
        }

        .xs {
          flex-grow: 1;
          max-width: 100%;
          flex-basis: 0;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }
      </style></div><div class=\\"item  mock \\"><div class=\\"item  mock \\">test<style>
        .item {
        }

        .justify {
          justify-content: undefined;
        }

        .direction {
          flex-direction: undefined;
        }

        .alignContent {
          align-content: undefined;
        }

        .alignItems {
          align-items: undefined;
        }

        .xs {
          flex-grow: 1;
          max-width: 100%;
          flex-basis: 0;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }
      </style></div><div class=\\"item  mock \\"><div class=\\"item  mock \\">test<style>
        .item {
        }

        .justify {
          justify-content: undefined;
        }

        .direction {
          flex-direction: undefined;
        }

        .alignContent {
          align-content: undefined;
        }

        .alignItems {
          align-items: undefined;
        }

        .xs {
          flex-grow: 1;
          max-width: 100%;
          flex-basis: 0;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }
      </style></div><div class=\\"item  mock \\"><div class=\\"item  mock \\">test<style>
        .item {
        }

        .justify {
          justify-content: undefined;
        }

        .direction {
          flex-direction: undefined;
        }

        .alignContent {
          align-content: undefined;
        }

        .alignItems {
          align-items: undefined;
        }

        .xs {
          flex-grow: 1;
          max-width: 100%;
          flex-basis: 0;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }
      </style></div><style>
        .item {
        }

        .justify {
          justify-content: undefined;
        }

        .direction {
          flex-direction: undefined;
        }

        .alignContent {
          align-content: undefined;
        }

        .alignItems {
          align-items: undefined;
        }

        .xs {
          flex-grow: 1;
          max-width: 100%;
          flex-basis: 0;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }
      </style></div>,<style>
        .item {
        }

        .justify {
          justify-content: undefined;
        }

        .direction {
          flex-direction: undefined;
        }

        .alignContent {
          align-content: undefined;
        }

        .alignItems {
          align-items: undefined;
        }

        .xs {
          flex-grow: 1;
          max-width: 100%;
          flex-basis: 0;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }
      </style></div>,<style>
        .item {
        }

        .justify {
          justify-content: undefined;
        }

        .direction {
          flex-direction: undefined;
        }

        .alignContent {
          align-content: undefined;
        }

        .alignItems {
          align-items: undefined;
        }

        .xs {
          flex-grow: 1;
          max-width: 100%;
          flex-basis: 0;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }
      </style></div>,<style>
        .item {
        }

        .justify {
          justify-content: undefined;
        }

        .direction {
          flex-direction: undefined;
        }

        .alignContent {
          align-content: undefined;
        }

        .alignItems {
          align-items: undefined;
        }

        .xs {
          flex-grow: 1;
          max-width: 100%;
          flex-basis: 0;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }
      </style></div>"
`;

exports[`Grid should render correctly 1`] = `
"<div class=\\"item  mock \\"><div class=\\"item  mock \\">test<style>
        .item {
        }

        .justify {
          justify-content: undefined;
        }

        .direction {
          flex-direction: undefined;
        }

        .alignContent {
          align-content: undefined;
        }

        .alignItems {
          align-items: undefined;
        }

        .xs {
          flex-grow: 1;
          max-width: 100%;
          flex-basis: 0;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }
      </style></div><div class=\\"item  mock \\">test<style>
        .item {
        }

        .justify {
          justify-content: undefined;
        }

        .direction {
          flex-direction: undefined;
        }

        .alignContent {
          align-content: undefined;
        }

        .alignItems {
          align-items: undefined;
        }

        .xs {
          flex-grow: 1;
          max-width: 100%;
          flex-basis: 0;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }
      </style></div><style>
        .item {
        }

        .justify {
          justify-content: undefined;
        }

        .direction {
          flex-direction: undefined;
        }

        .alignContent {
          align-content: undefined;
        }

        .alignItems {
          align-items: undefined;
        }

        .xs {
          flex-grow: 1;
          max-width: 100%;
          flex-basis: 0;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }
      </style></div>"
`;

exports[`Grid should work correctly when size exceeds 1`] = `
"<div class=\\"item  mock \\"><div class=\\"item xs mock \\">test<style>
        .item {
        }

        .justify {
          justify-content: undefined;
        }

        .direction {
          flex-direction: undefined;
        }

        .alignContent {
          align-content: undefined;
        }

        .alignItems {
          align-items: undefined;
        }

        .xs {
          flex-grow: 0;
          max-width: 100%;
          flex-basis: 100%;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 0;
            max-width: 100%;
            flex-basis: 100%;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }
      </style></div><div class=\\"item xs mock \\">test<style>
        .item {
        }

        .justify {
          justify-content: undefined;
        }

        .direction {
          flex-direction: undefined;
        }

        .alignContent {
          align-content: undefined;
        }

        .alignItems {
          align-items: undefined;
        }

        .xs {
          flex-grow: 0;
          max-width: 0;
          flex-basis: 0;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 0;
            max-width: 0;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }
      </style></div><style>
        .item {
        }

        .justify {
          justify-content: undefined;
        }

        .direction {
          flex-direction: undefined;
        }

        .alignContent {
          align-content: undefined;
        }

        .alignItems {
          align-items: undefined;
        }

        .xs {
          flex-grow: 1;
          max-width: 100%;
          flex-basis: 0;
          
        }

        @media only screen and (max-width: 650px) {
          .xs {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 650px) {
          .sm {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 900px) {
          .md {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1280px) {
          .lg {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }

        @media only screen and (min-width: 1920px) {
          .xl {
            flex-grow: 1;
            max-width: 100%;
            flex-basis: 0;
            
          }
        }
      </style></div>"
`;
